src/pymorize/std_lib/__init__.py

```diff
@@ -0,0 +1,323 @@
+
+
+def load_data(
+    data: Union[DataArray, Dataset, None], rule: Rule
+) -> Union[DataArray, Dataset]:
+    """
+    Load data from files according to the rule specification.
+
+    This function opens and combines data from multiple files that match the pattern
+    specified in the rule. It's useful for loading time series data that may be
+    spread across multiple files.
+
+    Parameters
+    ----------
+    data : xarray.DataArray or xarray.Dataset or None
+        Existing data (if any) to incorporate with loaded data.
+    rule : Rule
+        The rule containing the input patterns and other specifications
+        for loading the data.
+
+    Returns
+    -------
+    xarray.DataArray or xarray.Dataset
+        The loaded data combined into a single Dataset or DataArray.
+
+    Notes
+    -----
+    The rule_spec dictionary should contain an "input_patterns" key with a list
+    of file patterns to match, e.g., ["path/to/data/*.nc"].
+    """
+    return _load_data(data, rule)
+
+
+def get_variable(
+    data: Union[DataArray, Dataset], rule: Rule
+) -> Union[DataArray, Dataset]:
+    """
+    Extract a variable from a dataset as a DataArray.
+
+    Parameters
+    ----------
+    data : xarray.Dataset
+        The dataset containing the variable to extract.
+    rule : Rule
+        The rule containing the variable name to extract.
+
+    Returns
+    -------
+    xarray.DataArray
+        The extracted variable as a DataArray.
+
+    Raises
+    ------
+    KeyError
+        If the variable specified in the rule does not exist in the dataset.
+    """
+    if isinstance(data, Dataset):
+        variable_name = rule.model_variable
+        if variable_name not in data:
+            raise KeyError(f"Variable '{variable_name}' not found in dataset")
+        return data[variable_name]
+    return data
+
+
+def temporal_resample(
+    data: Union[DataArray, Dataset], rule: Rule
+) -> Union[DataArray, Dataset]:
+    """
+    Resample a DataArray or Dataset to a different temporal frequency.
+
+    Parameters
+    ----------
+    data : xarray.DataArray or xarray.Dataset
+        The data to resample.
+    rule : Rule
+        The rule containing parameters for the resampling operation,
+        including the frequency for resampling.
+
+    Returns
+    -------
+    xarray.DataArray or xarray.Dataset
+        The resampled data.
+
+    Notes
+    -----
+    This function resamples time series data to a different frequency.
+    The frequency is determined from the rule (typically from data_request_variable.frequency).
+    Common frequencies include:
+    - 'YS': year start
+    - 'MS': month start
+    - 'D': daily
+    - 'H': hourly
+
+    The resampling method (mean, sum, etc.) is determined by the rule.
+    """
+    if not has_time_axis(data):
+        return data
+
+    time_dim = get_time_label(data)
+    freq = rule.data_request_variable.frequency
+    return data.resample({time_dim: freq}).mean()
```

Comment:
What happens if the resampling frequency is higher than the data's frequency? Does `resample` raises an error?

Reply:
I'm not sure, to be honest. I'll make a separate issue for that and build a test so we can check that. 

We have two options, either:

A. Do whatever xarray does natively (this is the current implementation)
B. Check the frequency requested vs. data and report errors. I like this one better.

Reply:
See #133 for further discussion.